Client/Impl/Misc/PersistedAccessTokenCache.cs

```diff
@@ -86,7 +111,17 @@ private async Task<AccessToken> FetchNewAccessToken(string audience)
     {
         var newAccessToken = await accessTokenFetcherAsync();
         CachedCredentials[audience] = newAccessToken;
-        WriteCredentials();
+
+        await semaphore.WaitAsync();
```

Comment:
❓ I guess we could use just [lock](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock) here right? Any specifics/benefits about the semphore here (as you also use just one count)? Please be aware coming from the java world, so I might not be aware of some language specifics from c#.

Reply:
I will check if I can use just a lock. imo I can block the thread and wait until its available with lock. But in current case I want to check from another thread if the lock is in place and then it would not wait to fetch another token. Rather it could use existing token if its still valid.
There is some [Monitor](https://learn.microsoft.com/en-us/dotnet/api/system.threading.monitor?view=net-8.0) that is using lock behind the scene. 
I will mark this PR as draft for now and come back with the updates.

Reply:
Thanks that makes sense 👍